react/features/base/participants/actions.ts

```diff
@@ -253,6 +254,42 @@ export function participantJoined(participant: IParticipant) {
     };
 }
 
+/**
+ * Updates the sources of a remote participant.
+ *
+ * @param {IJitsiParticipant} jitsiParticipant - The IJitsiParticipant instance.
+ * @returns {{
+ *      type: PARTICIPANT_SOURCES_UPDATED,
+ *      participant: IParticipant
+ * }}
+ */
+export function participantSourcesUpdated(jitsiParticipant: IJitsiParticipant) {
+    return (dispatch: IStore['dispatch'], getState: IStore['getState']) => {
+        if (!jitsiParticipant) {
+            return;
+        }
+        const id = jitsiParticipant.getId();
+        const participant = getParticipantById(getState(), id);
+
+        if (participant?.local) {
+            return;
+        }
+        const sources = jitsiParticipant.getSources();
+
+        if (!sources?.size) {
+            return;
+        }
+
+        return dispatch({
+            type: PARTICIPANT_SOURCES_UPDATED,
```

Comment:
With ssrc-rewriting, updating the source-info in redux as soon as the presence for remote participant is received is very important since the bridge will start forwarding the media only when the source-info gets added to the receiver constraints. `PARTICIPANT_UPDATED` updates the state in redux after executing a bunch of stuff in the middleware (not related to ssrc-rewriting) and can possibly introduce delays.

Reply:
Sounds good.